src/analysis/refactor_open.ml

```diff
@@ -30,16 +30,16 @@ let qual_or_unqual_path mode ~open_lident ~open_path node_path node_lid =
       ->
       s :: acc
     | Pdot (path', s) when
-        mode = `Qualify && s = leftmost_open_lident ->
+        mode = `Qualify && s = open_leftmost_lident ->
       s :: acc
     | Pdot (path', s) ->
-      aux (s :: acc) path'
+      make_new_node_lid (s :: acc) path'
     | _ -> raise Not_found
   in
-  aux [] node_path
-
-let same_longident new_lident old_lident =
-  List.length new_lident = List.length (Longident.flatten old_lident)
+  let new_node_lid = make_new_node_lid [] node_path in
+  if String.equal node_leftmost_lident (List.hd new_node_lid) (* if new lid = old lid *)
+  then None
+  else Some (String.concat ~sep:"." new_node_lid)
```

Comment:
Since we're now returning an option, could we rewrite this to
```suggestion
match make_new_node_lid [] node_path with
| new_node_lid when not (String.equal node_leftmost_lident (List.hd new_node_lid)) ->
  Some (String.concat ~sep:"." new_node_lid)
| _ | exception Not_found -> None
```
So this function doesn't leak an exception (and we can simplify its caller).